lib/internal/http2/core.js

```diff
@@ -983,22 +986,79 @@ function emitClose(self, error) {
   self.emit('close');
 }
 
-function finishSessionDestroy(session, error) {
+function cleanupSession(session) {
   const socket = session[kSocket];
-  if (!socket.destroyed)
-    socket.destroy(error);
-
   session[kProxySocket] = undefined;
   session[kSocket] = undefined;
   session[kHandle] = undefined;
   session[kNativeFields] = new Uint8Array(kSessionUint8FieldCount);
-  socket[kSession] = undefined;
-  socket[kServer] = undefined;
+  if (socket) {
+    socket[kSession] = undefined;
+    socket[kServer] = undefined;
+  }
+}
+
+function finishSessionDestroy(session, error) {
+  debugSessionObj(session, 'finishSessionDestroy');
+
+  const handle = session[kHandle];
+  const socket = session[kSocket];
+  if (handle) handle.ondone = null;
+  cleanupSession(session);
+
+  if (socket && !socket.destroyed)
+    socket.destroy(error);
```

Comment:
NOTE, `socket.destroy` could schedule an error *after* the `emitClose` below. I'm not exactly sure what happens on socket error here.

Reply:
This part stayed the same as before. I think an error will be caught by `socketOnError` and ignored because `socket[kSession]` is now `undefined`.

Reply:
Ah, I see. Next question though. Should that error be ignored? There is no point in passing `error` to `destroy(error)` if it's going to be ignored anyway.

Reply:
I guess it's for 'correctness' reasons only, like 'we destroy the socket because of this error'.
As for the consequent errors, I think it's fine to ignore, we have already closed the session (and possibly even destroyed) so we don't care about the socket anymore as long as it closes.